map-storage/src/Services/CustomEntityCollectionService.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    CollisionGrid,
+    ENTITIES_FOLDER_PATH,
+    ENTITY_COLLECTION_FILE,
+    EntityCollectionRaw,
+    EntityRawPrefab,
+    mapCustomEntityDirectionToDirection,
+} from "@workadventure/map-editor";
+import { DeleteCustomEntityMessage, ModifyCustomEntityMessage, UploadEntityMessage } from "@workadventure/messages";
+import { fileSystem } from "../fileSystem";
+import { mapPathUsingDomainWithPrefix } from "./PathMapper";
+
+export class CustomEntityCollectionService {
+    private readonly hostname: string;
+
+    constructor(hostname: string) {
+        this.hostname = hostname;
+    }
+
+    private getEntityCollectionFileVirtualPath() {
+        return mapPathUsingDomainWithPrefix(`${ENTITIES_FOLDER_PATH}/${ENTITY_COLLECTION_FILE}`, this.hostname);
+    }
+
+    private getEntityToUploadVirtualPath(fileName: string) {
+        return mapPathUsingDomainWithPrefix(`${ENTITIES_FOLDER_PATH}/${fileName}`, this.hostname);
+    }
+
+    public async uploadEntity(uploadEntityMessage: UploadEntityMessage) {
+        const { imagePath, file } = uploadEntityMessage;
+        await fileSystem.writeByteAsFile(this.getEntityToUploadVirtualPath(imagePath), file);
+        await this.addEntityInEntityCollectionFile(
+            this.mapEntityFromUploadEntityMessageToEntityRawPrefab(uploadEntityMessage)
+        );
+        return;
+    }
+
+    public async modifyEntity(modifyCustomEntityMessage: ModifyCustomEntityMessage) {
+        const { id, name, tags, depthOffset } = modifyCustomEntityMessage;
+        let collisionGrid = undefined;
+        if (modifyCustomEntityMessage.collisionGrid) {
+            collisionGrid = CollisionGrid.parse(modifyCustomEntityMessage.collisionGrid);
+        }
+        const customEntityCollectionFileContent = await this.readOrCreateEntitiesCollectionFile();
+        const customEntityCollection = EntityCollectionRaw.parse(JSON.parse(customEntityCollectionFileContent));
+        const indexOfEntityToModify = customEntityCollection.collection.findIndex((entity) => entity.id === id);
+        if (indexOfEntityToModify !== -1) {
```

Comment:
We should probably have an "else" statement here that logs an error and sends it to Sentry as not finding the entity we modify should almost never happen.